src/js/components/form-elements/v2/DateFieldInput.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { forwardRef } from 'react';
+
+import PropTypes from 'prop-types';
+import { RiCalendarLine, RiCloseLine } from 'react-icons/ri';
+
+import './style.scss';
+
+const DateFieldInput = forwardRef(({
+  value,
+  placeholder,
+  onClear,
+  className,
+  disabled,
+  ...props
+}, ref) => {
+  const disabledProps = disabled ? {
+    className: `disabled ${className}`,
+  } : {
+    className,
+    tabIndex: 0,
+    role: 'button',
+  };
```

Comment:
could you elaborate on this part?
if input is disabled we want to remove the `role: button` from the div wrapper and `tabIndex: 0`

Why do we even need these attributes here? is it for the ability to **Tab** click and navigate with a keyboard between inputs or is there something more to it?

Reply:
Divs are not focusable by default, so thanks to the `tabIndex: 0` and `role: button` I can access the `:focus` on the div element. When the input is disabled and the tabIndex and role are still present this div still can get a focus and something weird happens when trying to get focus on other date input elements.https://assets.grammarly.com/emoji/v1/1f64c.svg